ee/app/livechat-enterprise/server/lib/Helper.js

```diff
@@ -137,3 +138,31 @@ export const allowAgentSkipQueue = (agent) => {
 
 	return settings.get('Livechat_assign_new_conversation_to_bot') && hasRole(agent.agentId, 'bot');
 };
+
+export const setPredictedVisitorAbandonmentTime = (room) => {
+	if (!room.v || !room.v.lastMessageTs || !settings.get('Livechat_auto_close_abandoned_rooms')) {
+		return;
+	}
+
+	let secondsToAdd = settings.get('Livechat_visitor_inactivity_timeout');
+
+	const department = room.departmentId && LivechatDepartment.findOneById(room.departmentId);
+	if (department && department.visitorInactivityTimeoutInSeconds) {
+		secondsToAdd = department.visitorInactivityTimeoutInSeconds;
+	}
+
+	if (secondsToAdd <= 0) {
+		return;
+	}
+
+	const willBeAbandonedAt = moment(room.v.lastMessageTs).add(Number(secondsToAdd), 'seconds').toDate();
+	LivechatRooms.setPredictedVisitorAbandonment(room._id, willBeAbandonedAt);
+};
+
+export const updatePredictedVisitorAbandonment = (closeRooms) => {
+	if (closeRooms) {
```

Comment:
I think we should check this setting here: `settings.get('Livechat_auto_close_abandoned_rooms')`
We can avoid an unnecessary loop here.